src/analysis/souffle/tests/arcs_fact_tests/fail_check_on_subpaths.dl

```diff
@@ -22,8 +22,8 @@
 #include "taint.dl"
 #include "fact_test_helper.dl"
 
-claimHasTag("R.P1.foo.a", "userSelection").
-claimHasTag("R.P1.foo.b", "screenContent").
+claimHasTag("P1", "R.P1.foo.a", "userSelection").
```

Comment:
I have two reasons that I'm referencing the `ParticleSpec` here, one dubious and philosophical, the other practical.

First, the dubious, philosophical one: The way I was thinking about this was that a `Principal` is associated with a particular `ParticleSpec`, not a particular `Particle`. The reason why I was thinking of it that way was that I was assuming that all code contained in a particular `ParticleSpec` was written by the same people and thus would behave the same way. However, as soon as I started working with the examples with hardware that we discussed, I realized that a `ParticleSpec` defines only an interface, and thus may be used for multiple different `Principal`s which may behave very differently.

Now, the practical reason, which may have caused me to unconciously perform a bit of motivated reasoning towards the philosophical one: `Particle` instances don't have names that can be directly referenced at static time. We generate `Particle` names for `AccessPath`s, but to refer to such a name from the authorization logic would require the person writing the authorization logic to know the details of how we generate them, which seems like a bad idea.

So I think that it would make sense to eventually reference `Particle`s in these claims, but I don't think we can do so in a way that acts as a `JOIN` key with authorization logic with the current Arcs manifest IR.

Reply:
Your philosophical reasoning makes sense: particle spec expresses behavior of all instances of the spec. Whether something is a `tag` or not should be explicitly stated.